### ISSUE TITLE
Superficial restrictions on DB access

### ISSUE THREAD

**Issue by b1bart** [EOSIO/spec-repo#3]
### Concern

Refer to #2 as well which suggests we remove the conflation of where the data is stored.

The database enhancement proposal lists the following motivation: 

https://github.com/EOSIO/eep-proposal-staging/blob/b2eaab8e5e41d431fa9c6c60ffd05db9bc340ae8/eep-draft_enhanced_database.md#L35

#### Broader Issue

While this is a true statement, it does not represent the full complexity of that issue which includes the potential effects of such upgrades on indexers, explorers, wallets and other user-interfaces (via SHiP, demux, etc).  

#### Impact to Use cases

Meanwhile, there is no discussion of the valuable use-cases that removing this read-only access also removes such as supporting convenient access to "registries" or "caches" of data stored in this flexible database model.  

For instance, using the communities implementation of vRAM or light-clients but backing this data in a KV store instead currently depends on direct data access.  The ABI of this data access is negotiated between contracts and the platform level should not feel responsible for dictating how this negotiation is performed or what its result is.  

#### Developer confusion 

Futhermore, this presents a false dichotomy to the developer, if they want to support cross-contract reads they must use the table interface despite there being no real reason to require that consideration.  Because of this, the developer is not free to review the available data access models and chose the one best suited to their data model, they must be aware of the extra limitations/capabilities of the models and the potentially rippling effects their choice of data access model will have.

TL;DR removing cross contract data reads seems to solve a small part of the perceived problem and creates a larger cost to use-cases and cognitive load for developers as their choice of access model is encumbered by unrelated concerns.
